File: 2024/202. Happy Number.cpp
class Solution {
public:
    bool isHappy(int n) {
        unordered_set<int> set;
        while (n != 1) {
            set.insert(n);
            int next = 0;
            while (n != 0) {
                next += (n % 10) * (n % 10);
                n /= 10;
            }
            n = next;
            if (set.find(n) != set.end()) {
                return false;
            }
        }
        return true;
    }
};

File: 2024/202. Happy Number.py
class Solution:
    def isHappy(self, n: int) -> bool:
        seen = set()
        while n != 1 and n not in seen:
            seen.add(n)
            next = 0
            while n != 0:
                next += (int)(n % 10) ** 2
                n /= 10
            n = next
        return n == 1
